test/spec/features/planes/PlanesBehaviorSpec.js

```diff
@@ -71,6 +105,40 @@ describe('features/planes/PlanesBehavior', function() {
       expect(canvas.getActivePlane().name).to.equal('base');
     }));
 
+
+    it('should work after re-adding a plane', inject(function(canvas, modeling, commandStack) {
+
+      // given
+      var plane = canvas.createPlane('1');
+      canvas.setActivePlane('1');
+
+      var shape2 = canvas.addShape({
+        id: 'shape2',
+        x: 10,
+        y: 10,
+        width: 100,
+        height: 100
+      });
+      modeling.removeShape(shape2);
+
+      // remove and re-add the plane, e.g. when removing a subprocess and undoing it
+      canvas.setRootElementForPlane(null, plane, true);
```

Comment:
What we're doing here is essentially operating on two different layers: Doing a modeling operation on the one hand, and doing low level operations (testing exactly what?) on the other hand.

What I'd personally like here is to create a simple test command that executes a real world use-case, i.e. do/undo/redo of adding plane + switching to it, adding an element and so forth. Then we have something that works end-to-end and can be tested in an integration test manner.

Reply:
Example:

* Implement a behavior that creates a plane any time an element is being created
* Drill into that plane, add a task
* Undo both things (and see how active planes change); re-do both things (and again, verify what is happening).